artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AmqpCoreConverter.java

```diff
@@ -345,6 +346,12 @@ private static ServerJMSMessage processProperties(ServerJMSMessage jms, Properti
             }
          }
          Object correlationID = properties.getCorrelationId();
+
+         if (properties.getMessageId() != null) {
+            //core jms clients get JMSMessageID from UserID which is a UUID object
+            jms.getInnerMessage().setUserID(UUIDGenerator.getInstance().generateUUID());
```

Comment:
If the Core bits truly use UUID here then one thing you could do is attempt to preserve the original ID by checking if it is a UUID as that is one of the legal Message ID types:
http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-message-id-uuid

By preserving a true UUID as message ID you'd also be reducing the cost of the conversion by avoiding string operations to create the UUID string and also to encode the string to UTF8 for on the wire vs sending a couple longs and a byte to represent the value.

Reply:
Looks like proton-j maps to java UUID, but the core message using it's own UUID type.
If we preserve the original uuid, we need to convert the type. I'll take a look at the code and what need to be changed.